### PR TITLE
Adds colony info to the debug screen + adds some sonar fixes.

### LINKED ISSUE #163: Add closest colony to debug screen
I think it would be easy to show the closest colonies in all 4 direction to the player in the f3 screen.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #163

# Changes proposed in this pull request:
- Adds colony info to debug screen
- fixes some sonar issues
-

Review please @Minecolonies/maintainers

### CODE REVIEW

File: src/main/java/com/minecolonies/colony/IColony.java [hunk 1]
```diff
@@ -4,51 +4,54 @@
...
     IPermissions getPermissions();
 
     /**
-     * Determine if a given chunk coordinate is considered to be within the colony's bounds
+     * Determine if a given chunk coordinate is considered to be within the colony's bounds.
      *
-     * @param w   World to check
-     * @param pos Block Position
-     * @return True if inside colony, otherwise false
+     * @param w   World to check.
```
Comment (Kostronor) [inline src/main/java/com/minecolonies/colony/IColony.java]: I'd say leave the dots from parameters :p

File: src/main/java/com/minecolonies/event/EventHandler.java [hunk 2]
```diff
@@ -29,6 +33,45 @@
...
+            IColony colony = ColonyManager.getIColony(world,player.getPosition());
+            final double minDistance = ColonyManager.getMinimumDistanceBetweenTownHalls();
+
+            if(colony == null)
+            {
+                colony = ColonyManager.getClosestIColony(world, player.getPosition());
+
+                if (colony == null || Math.sqrt(colony.getDistanceSquared(player.getPosition())) > 2 * minDistance)
+                {
+                    event.getLeft().add(LanguageHandler.format("com.minecolonies.gui.debugScreen.noCloseColony"));
```
Comment (Kostronor) [inline src/main/java/com/minecolonies/event/EventHandler.java]: So mindistance text is missing when there is no other colony near?

File: src/main/resources/assets/minecolonies/lang/en_US.lang [hunk 3]
```diff
@@ -179,3 +179,8 @@ com.minecolonies.gui.workerHuts.farmerHut.unused=unused
 com.minecolonies.gui.scarecrow.user=Field used by: %s
 com.minecolonies.gui.scarecrow.user.noone=no-one
 com.minecolonies.gui.buildtool.nohutininventory=The hut was not found in your inventory.
+
+com.minecolonies.gui.debugScreen.noCloseColony=No close colony!
+com.minecolonies.gui.debugScreen.nextColony=Next Colony in: %d.
+com.minecolonies.gui.debugScreen.blocksFromCenter=%d blocks from the center.
+com.minecolonies.gui.debugScreen.required=%d required to place a Colony.
```
Comment (Kostronor) [inline src/main/resources/assets/minecolonies/lang/en_US.lang]: This text tells me nothing imo. 86 Blocks maybe?

File: src/main/resources/assets/minecolonies/lang/en_US.lang [hunk 4]
```diff
@@ -179,3 +179,8 @@ com.minecolonies.gui.workerHuts.farmerHut.unused=unused
 com.minecolonies.gui.scarecrow.user=Field used by: %s
 com.minecolonies.gui.scarecrow.user.noone=no-one
 com.minecolonies.gui.buildtool.nohutininventory=The hut was not found in your inventory.
+
+com.minecolonies.gui.debugScreen.noCloseColony=No close colony!
+com.minecolonies.gui.debugScreen.nextColony=Next Colony in: %d.
```
Comment (Kostronor) [inline src/main/resources/assets/minecolonies/lang/en_US.lang]: Next colony is x blocks away

File: src/main/resources/assets/minecolonies/lang/en_US.lang (hunk 3 above)
Comment (Raycoms) [inline src/main/resources/assets/minecolonies/lang/en_US.lang]: The text will be: Next colony in: %d (%d required to place a colony)
Blocks would be doubled

File: src/main/java/com/minecolonies/event/EventHandler.java (hunk 2 above)
Comment (Raycoms) [inline src/main/java/com/minecolonies/event/EventHandler.java]: We don't need the mindistance text if we're able to place our colony right away.

File: src/main/java/com/minecolonies/colony/IColony.java (hunk 1 above)
Comment (Raycoms) [inline src/main/java/com/minecolonies/colony/IColony.java]: But it's so unsatisfying to have no end there.
I always put a dot there.

File: src/main/resources/assets/minecolonies/lang/en_US.lang (hunk 3 above)
Comment (Kostronor) [inline src/main/resources/assets/minecolonies/lang/en_US.lang]: Next colony is %d blocks away (%d required to place a colony)

File: src/main/java/com/minecolonies/event/EventHandler.java (hunk 2 above)
Comment (Kostronor) [inline src/main/java/com/minecolonies/event/EventHandler.java]: It's a nice info to have always visible imo

File: src/main/java/com/minecolonies/colony/IColony.java (hunk 1 above)
Comment (Kostronor) [inline src/main/java/com/minecolonies/colony/IColony.java]: Alright :D

File: src/main/java/com/minecolonies/event/EventHandler.java [hunk 5]
```diff
@@ -29,6 +36,49 @@
...
+                {
+                    colony = ColonyManager.getClosestIColony(world, player.getPosition());
+
+                    if (colony == null || Math.sqrt(colony.getDistanceSquared(player.getPosition())) > 2 * minDistance)
+                    {
+                        event.getLeft().add(LanguageHandler.format("com.minecolonies.gui.debugScreen.noCloseColony"));
+                        return;
+                    }
+
+                    event.getLeft().add(LanguageHandler.format("com.minecolonies.gui.debugScreen.nextColony", (int) Math.sqrt(colony.getDistanceSquared(player.getPosition())))
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/event/EventHandler.java]: Why is this two separate strings? Why not just one?

File: src/main/resources/assets/minecolonies/lang/en_US.lang [hunk 6]
```diff
@@ -179,3 +179,8 @@ com.minecolonies.gui.workerHuts.farmerHut.unused=unused
 com.minecolonies.gui.scarecrow.user=Field used by: %s
 com.minecolonies.gui.scarecrow.user.noone=no-one
 com.minecolonies.gui.buildtool.nohutininventory=The hut was not found in your inventory.
+
+com.minecolonies.gui.debugScreen.noCloseColony=No close colony!
+com.minecolonies.gui.debugScreen.nextColony=Next colony is %d blocks away.
```
Comment (cltnschlosser) [inline src/main/resources/assets/minecolonies/lang/en_US.lang]: Could be "Next colony is %d blocks away (%d required to place a colony)", I don't think the periods are necessary.

**Comment by cltnschlosser** [comment]
Yea, as many as you want.

**Comment by Raycoms** [comment]
Changed it to one line, please approve @Minecolonies/maintainers

**Comment by Kostronor** [comment]
:shipit: @cltnschlosser

**Pull request merged**